### PR TITLE
Documentation for query

### PR DESCRIPTION
**Pull request by Amarcolina**
- Added basic documentation to all query ops
 - Fixed some out of date documentation
 - Fixed cases where the queries could get into an invalid state after you used them once
 - Added the Last and Single operations
 - Added copywrite headers to everything

### CODE REVIEW

File: Assets/LeapMotion/Scripts/Query/QueryWrapper.cs [hunk 1]
```diff
@@ -12,39 +12,45 @@
...
-  /// There is some difference between using Linq and using Leap queries. One is that is you must prefix your query with
-  /// a call to a Query() method if you are starting with an external data structure.  So for example if you want to query a
-  /// list, your method call would look something like this:
+  /// There is one big difference between using Linq and using Leap queries.You must prefix your query 
+  /// with a call to a Query() method if you are starting with an external data structure.  So for 
+  /// example if you want to query a list, your method call would look something like this:
   ///
   /// myList.Query().Where(someCondition).First();
-  ///
-  /// The other major difference is that you can only use a query once.  So the following will not work:
```
Comment (jselstad) [inline Assets/LeapMotion/Scripts/Query/QueryWrapper.cs]: Is this no longer the case?

File: Assets/LeapMotion/Scripts/Query/QueryWrapper.cs [hunk 2]
```diff
@@ -70,29 +76,76 @@ public struct Enumerator {
...
+  /// </summary>
   public interface IQueryOp<T> {
+
+    /// <summary>
+    /// Tries to get the next value in the sequence.  If this method returns true,
+    /// the next value will be placed into the out parameter t.  If this method
+    /// returns false, the sequence is at an end and t will be the default value
+    /// of T.
+    /// 
+    /// Once TryGetNext returns false, it can NEVER return true again until the
```
Comment (jselstad) [inline Assets/LeapMotion/Scripts/Query/QueryWrapper.cs]: Or has the warning been moved down here?

File: Assets/LeapMotion/Scripts/Query/QueryWrapper.cs (hunk 1 above)
Comment (Amarcolina) [inline Assets/LeapMotion/Scripts/Query/QueryWrapper.cs]: It is no longer the case!  You can now re-use queries as much as you want!

File: Assets/LeapMotion/Scripts/Query/QueryWrapper.cs (hunk 2 above)
Comment (Amarcolina) [inline Assets/LeapMotion/Scripts/Query/QueryWrapper.cs]: This is slightly different.  In the general case, you will never interact with TryGetNext directly so you never need to worry about calling Reset().  Normally you will either use a foreach(), or a direct op like First() / ToList() / ect... all of which do not need to be manually reset.

**Comment by Amarcolina** [comment]
I will add examples to each query op!

**Pull request merged**